### PR TITLE
Test change detection. Make more XML fields live.

### PR DESCRIPTION
**Pull request by awong-dev**
Start actually extracting data from the input fields. Makes 7 fields live.

Part of https://github.com/department-of-veterans-affairs/healthcare-application/issues/21

### CODE REVIEW

**Comment by awong-dev** [comment]
@akainic Willing do do some moar reviews? :D

File: _health-care/_js/_form.js [hunk 1]
```diff
@@ -27,258 +53,258 @@ var HealthApp = (function() {
...
-    { node: "ChildLiveNo",  arg: "0", transform: identityTransform },
-    { node: "ChildProvideSupportYes",  arg: "0", transform: identityTransform },
-    { node: "ChildProvideSupportNo",  arg: "0", transform: identityTransform },
-    { node: "ChildGrossIncome",  arg: "54.00", transform: identityTransform },
-    { node: "ChildPropertyIncome",  arg: "43.00", transform: identityTransform },
-    { node: "ChildOtherIncome",  arg: "32.00", transform: identityTransform }
+    { node: "DateOfBirthMonth",  arg: 'veteran[date_of_birth]', transform: extractMonthTransform },
+    { node: "DateOfBirthDay",  arg: 'veteran[date_of_birth]', transform: extractDateTransform },
+    { node: "DateOfBirthYear",  arg: 'veteran[date_of_birth]', transform: extractYearTransform },
+    { node: "PlaceOfBirthCity",  name: 'veteran[]', arg: "Kent", transform: identityTransform },
```
Comment (awong-dev) [inline _health-care/_js/_form.js]: Note, I'm using the `name: 'veteran[]'` field as a tag to know what I've looked at and what I haven't. It also lets me do a regex to auto-generate the unittest auto-fill lines. They will go away when we're fully done here.

File: _health-care/_js/_form.js [hunk 2]
```diff
@@ -1,21 +1,47 @@
 var HealthApp = (function() {
   "use strict";
 
-  var identityTransform = function(formData, arg) { return arg; };
-  var extractTransform = function(formData, arg) { return formData[arg]; };
+  function identityTransform(formData, arg) { return arg; }
+  function extractTransform(formData, arg) { return formData[arg]; }
```
Comment (akainic) [inline _health-care/_js/_form.js]: For my own understanding: why did you change the way you're declaring the functions here?

File: _health-care/_personal_information/_name_and_general_information.html [hunk 3]
```diff
@@ -48,11 +48,20 @@
 
     <div class="row">
       <div class="small-3 columns">
+        <label for="veteran_date_of_birth">Date of Birth</label>
+      </div>
+      <div class="small-9 columns">
+        <input type="text" name="veteran[date_of_birth]" />
```
Comment (akainic) [inline _health-care/_personal_information/_name_and_general_information.html]: This probably isn't the best input type for a date field, as you'll probably get all kinds of weird formats and submissions (unless you want to include some sort of client-side validation before anything is submitted). HTML5 has a `input type="date"`, but that's not supported in IE yet. Rails has a form helper called `date_select`, which creates 3 individual dropdowns (month, day, year), and you could use that with form_for. But we could probably wait to implement that until later.

File: _health-care/_personal_information/_name_and_general_information.html [hunk 4]
```diff
@@ -48,11 +48,20 @@
...
+    <div class="row">
+      <div class="small-3 columns">
         <label for="veteran_gender">Gender</label>
       </div>
       <div class="small-9 columns">
-        <select name="veteran[gender]" ><option value="0">Male</option>
-          <option value="1">Female</option>
+        <select name="veteran[gender]" ><option value="M">Male</option>
+          <option value="F">Female</option>
           <option value="2">NeedOtherOfficialOptions</option></select>
```
Comment (akainic) [inline _health-care/_personal_information/_name_and_general_information.html]: Do you still want to keep this value at `2` or do you want to change it to `O` or something to be more consistent with the naming of the other values?

**Comment by akainic** [comment]
LGTM!

File: _health-care/_personal_information/_name_and_general_information.html (hunk 4 above)
Comment (awong-dev) [inline _health-care/_personal_information/_name_and_general_information.html]: All the select/radio boxes are pretty much wrong as I was just mocking up the UI when I first did this. There needs to be a pass done to audit all of them so this is just a note to myself.  It made more sense in the rails code that generated it...

File: _health-care/_personal_information/_name_and_general_information.html (hunk 3 above)
Comment (awong-dev) [inline _health-care/_personal_information/_name_and_general_information.html]: Good catch! I will go with HTML5 date for now and we can do a polyfill later.

File: _health-care/_js/_form.js (hunk 2 above)
Comment (awong-dev) [inline _health-care/_js/_form.js]: function(foo) {} <-- creates an anonymous function that conceptually takes a closure of all the variables visible at that time. That closure uses memory itself and also keeps those variables alive. These little transformers don't actually reference anything outside their own scope so it probably actually gets optimized out, but I didn't want to rely on it.

Therefore I went back to bare named functions  (function iHaveAName(foo) {}) which does not create a closure.

**Comment by awong-dev** [comment]
Fixed comments! PTAL?

File: _health-care/_js/_form.js (hunk 2 above)
Comment (awong-dev) [inline _health-care/_js/_form.js]: ...though now that I've written that, I actually realized I don't know where I got this idea from. I will look at that later today to confirm.

Comment (akainic) [inline _health-care/_js/_form.js]: Ah okay, that makes sense, thanks!

Comment (awong-dev) [inline _health-care/_js/_form.js]: And by later today...I meant right now.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures
http://stackoverflow.com/questions/4071292/may-function-declarations-appear-inside-statements-in-javascript

So I'm wrong in 2 counts.
  1. Normal FunctionDeclarations (function iHaveAName() {}) do indeed create a closure if necessary.
  2. FunctionDeclarations are not allowed inside a block (anything surrounded by a {}); only Function Expressions are allowed. This is rooted in ECMAScript spec 5 so it's definitely authoritative. The fact that it works is actually a commonly implemented extension to JS.

Oops. I will fix in another PR cause I have a meeting to prep for right now.

File: _health-care/_personal_information/_name_and_general_information.html (hunk 4 above)
Comment (akainic) [inline _health-care/_personal_information/_name_and_general_information.html]: :+1:

File: _health-care/_js/_form.js (hunk 2 above)
Comment (akainic) [inline _health-care/_js/_form.js]: Oh interesting, so since you scoped all of this within `var HealthApp = function() {}`, function declarations technically aren't allowed? Learning new things!

Comment (awong-dev) [inline _health-care/_js/_form.js]: that's my reading of it...

This is all cause ES5 and earlier don't have any well defined module system so people are hacking in features using functions. It's ironic that JS, the butt of language jokes, is using all the techniques developed by LISP, the common ivory tower of language purists. :-/

Anyways, ES6 seems to have solved this by making it so top-level functions aren't in global (window?) scope by default which makes them way more useful.

http://www.sitepoint.com/understanding-es6-modules/

**Comment by awong-dev** [comment]
mergeing per prior LGTM.
